From 66d99f1f4c1101f0d9615d94d05104ea8a59fb6b Mon Sep 17 00:00:00 2001
From: Eneas U de Queiroz <cote2004-github@yahoo.com>
Date: Thu, 14 Jun 2018 18:30:55 -0300
Subject: [PATCH] mrib.c: use __constant_cpu_to_be32

gcc 7.3.0 thinks that cpu_to_be32() is not a compile-time constant, so
it does not allow it to initialize ipv4_rtr_alert outside of a
function.  Se we need to use __constant_cpu_to_be32 instead.

Signed-off-by: Eneas U de Queiroz <cote2004-github@yahoo.com>
---
 src/mrib.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/mrib.c b/src/mrib.c
index 5e0210b..c33abc5 100644
--- a/src/mrib.c
+++ b/src/mrib.c
@@ -54,7 +54,7 @@ struct mrib_iface {
 	struct uloop_timeout timer;
 };
 
-static uint32_t ipv4_rtr_alert = cpu_to_be32(0x94040000);
+static uint32_t ipv4_rtr_alert = __constant_cpu_to_be32(0x94040000);
 static struct {
 	struct ip6_hbh hdr;
 	struct ip6_opt_router rt;
-- 
2.16.4

